File: solutions/elixir/050-climbing-stairs.ex
# Problem 50: Climbing Stairs (LeetCode 70)
# Difficulty: Easy
# Language: Elixir
# 
defmodule ClimbStairs do
  def climb_stairs(n) do
    {result, _} = Enum.reduce(1..n, {1, 1}, fn _, {a, b} -> {b, a + b} end)
    result
  end
end

File: solutions/typescript/050-climbing-stairs.ts
/**
 * Problem 50: Climbing Stairs (LeetCode 70)
 * Difficulty: Easy
 * Language: TypeScript
 */
function climbStairs(n: number): number {
  return Array.from({ length: n }).reduce<[number, number]>(
    ([a, b]) => [b, a + b],
    [1, 1]
  )[0];
}
